docs/intro/upgrading-to-ionic-6.md

```diff
@@ -0,0 +1,443 @@
+
+### Icon
+
+Ionic 6 now ships with Ionicons 6. Review the [Ionicons 6 Breaking Changes Guide](https://github.com/ionic-team/ionicons/releases/tag/v6.0.0) and make any necessary changes.
+
+### Input
+
+Ensure `null` is not passed in as a value to the `placeholder` property. We recommend using `undefined` instead.
+
+### Modal
+
+`ion-modal` now uses the Shadow DOM. Update any styles targeting the internals of `ion-modal` to use either the [ion-modal CSS Variables](../api/modal#css-custom-properties) or the [ion-modal CSS Shadow Parts](../api/modal#css-shadow-parts):
+
+**Before**
+```css
+ion-modal .modal-wrapper {
+  ...
+}
+
+ion-modal ion-backdrop {
+  ...
+}
+```
+
+**After**
+```css
+ion-modal::part(content) {
+  ...
+}
+
+ion-modal::part(backdrop) {
+  ...
+}
+```
+
+### Popover
+
+`ion-popover` now uses the Shadow DOM. Update any styles targeting the internals of `ion-popover` to use either [ion-popover CSS Variables](../api/popover#css-custom-properties) or the [ion-popover CSS Shadow Parts](../api/popover#css-shadow-parts):
+
+
+**Before**
+```css
+ion-popover .popover-arrow {
+  ...
+}
+
+ion-popover ion-backdrop {
+  ...
+}
+
+ion-popover .popover-content {
+  ...
+}
+```
+
+**After**
+```css
+ion-popover::part(arrow) {
+  ...
+}
+
+ion-popover::part(backdrop) {
+  ...
+}
+
+ion-popover::part(content) {
+  ...
+}
+```
+
+### Radio
+
+Remove any usage of the `RadioChangeEventDetail` interface.
+
+### Select
+
+Ensure `null` is not passed in as a value to the `placeholder` property. We recommend using `undefined` instead.
+
+### Textarea
+
+Ensure `null` is not passed in as a value to the `placeholder` property. We recommend using `undefined` instead.
+
+### Browser Support
+
+The list of browsers that Ionic supports has changed. Review the [Browser Support Guide](../reference/browser-support) to ensure you are deploying apps to supported browsers.
+
+
+### Testing
+
+Ionic 6 now ships as ES Modules. ES Modules are supported in all major browsers and bring developer experience and code maintenance improvements. Developers testing with Jest will need to update their Jest configuration as Jest does not have full support for ES Modules as of Jest 27.
+
+This update involves using Babel to compile Ionic's ES Modules down to the CommonJS (CJS) format, a format that Jest can understand. Once Jest ships support for ES Modules, this change will no longer be necessary. See https://github.com/facebook/jest/issues/9430 for updates on ES Modules support in Jest.
+
+If you are starting fresh with a new Ionic app, this configuration is done for you in our starter applications. For those with existing Ionic apps, follow the steps below to get Jest working with Ionic 6:
+
+1. Add a `transformIgnorePatterns` field to your Jest config that includes the relevant Ionic packages. This is typically found in `jest.config.js` or the `jest` field in `package.json`:
+
+**jest.config.js**
+```js
+module.exports = {
+  ...,
```

Comment:
I would recommend we are consistent with:
```
...
```

without the trailing comma for both the jest config and package.json examples.